rfcs/RFC-Upload-From-Dropbox.md

```diff
@@ -0,0 +1,417 @@
+  will be updated. If the size of the file to be uploaded is known, the progress will indicate how much has been
+  uploaded over the total file size. If the total file size is not known, the progress will not be updated. This field is
+  returned using `GET submission/{submission_id}/upload`.
+- owner - the id of the user who owns this collection. This field is used to enforce if the user authorized to view the
+  remaining fields.
+- message - error message if the upload fails. This field is updated when an error occurs that should be visible to the
+  owner.
+
+#### Upload Bucket
+
+The upload bucket is a separate bucket. It is used to stage files
+before moving to the main bucket. Uploading files to a staging buckets makes it easier to clean up canceled, partial,
+or otherwise abandoned uploads, without accidentally deleting files actively being served by the Data Portal. Central
+Infra also provides a [bucket template](https://github.com/chanzuckerberg/cztack/tree/main/aws-s3-private-bucket) to help
+manage stale multipart uploads.
+
+S3 event can be used to trigger additional processing such as validation, once the upload has completed. The additional
+processing is outside the scope of this RFC.
+
+Files are uploaded to {collection_id}/{dataset_id}/{file_name} in the upload bucket.
+
+### Data Portal APIs
+
+The DP Browser App will use these endpoints to interact with the Backend using these APIs. This section describes the new
+endpoints and how they are used. These are the new API endpoints needed to complete an upload.
+
+These APIs replace the [PUT /v1/submission/{project_uuid}/file](https://docs.google.com/document/d/1d8tv2Ub5b3E7Il85adOAUcG8P05N6UBZJ3XbhJSRrFs/edit#heading=h.3hln6w2kzoyt)
+section.
+
+#### POST submission/{submission_id}/upload/link
+
+An authenticated user can upload a file from a shared link to a dataset in their submission.
+
+If the upload is in an error state, this endpoint can be used with the dataset_id to restart the submission. If a new
+link is provided, the new link will be used. If the submission is not in an error state, this endpoint will
+return an error.
+
+Starting an upload causes the database to be updated with a new dataset. The dataset will set the initial state of the upload
+to waiting once it has been accepted and is in the upload queue.
+
+**Request:**
+
+| Parameter             | Description                                                                    |
+| --------------------- | ------------------------------------------------------------------------------ |
+| Link                  | a shared link to the file                                                      |
+| submission_id         | identifies the submission                                                      |
+| _Optional_ dataset_id | identifies the dataset being uploaded. Used to check the status of the upload. |
```

Comment:
This can be deleted since the status is checked with the GET?

Reply:
This field is still used to restart the upload process if the upload failed.